### ISSUE TITLE
Add support for SAP HANA connection in `read_database` method. 

### ISSUE THREAD

**Issue by avhz** [pola-rs/polars#19729]
### Description

I am wondering if it is possible to add support for SAP HANA connections. 

The official client library is: https://pypi.org/project/hdbcli/

```python
from hdbcli import dbapi

connection = dbapi.connect(
    address="<hostname>",
    port=3<NN>MM,
    user="<username>",
    password="<password>"
)

query = "SELECT * FROM some_table"

df = polars.read_database(query, connection)
```

Currently will throw a `NotImplementedError`:

```python
NotImplementedError: Currently no support for 'hdbcli' connection <pyhdbcli.Cursor object at 0x7f3e03d3dc60>
```

An SQL Alchemy connection currently works as SAP provide their own [SQL Alchemy library](https://github.com/SAP/sqlalchemy-hana), but it might be faster to use their client library connection object directly ?

**Comment by alexander-beedie** [comment]
@avhz: I have just committed a tentative fix, but you'll have to validate it on your side as we don't have an SAP HANA instance to confirm it against. I _think_ it will work though 🤔

**Comment by avhz** [comment]
Awesome, thanks for the super quick turnaround ! 
I'l test it tomorrow and get back to. Cheers :)

**Comment by alexander-beedie** [comment]
> Awesome, thanks for the super quick turnaround ! I'l test it tomorrow and get back to you. Cheers :)

FYI: now available in the just-released `1.13.0`, so let me know how it goes 👌

**Comment by avhz** [comment]
Did a quick test this afternoon and it seems to work great! 
Thanks a lot for the patch, much appreciated :)
